python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -283,16 +280,9 @@ def _func_build_rf(
     def _fit(model, input_data, convert_dtype):
         X = input_data[0][0]
         y = input_data[0][1]
-        if isinstance(X, cudf.core.DataFrame):
-            if len(input_data) > 1:
-                for i in range(1, len(input_data)):
-                    X = cudf.concat([X, input_data[i][0]])
-                    y = cudf.concat([y, input_data[i][1]])
-        else:
-            X_parts = list(map(lambda x: cp.asarray(x[0]), input_data))
-            X = cp.concatenate(X_parts)
-            y_parts = list(map(lambda y: cp.asarray(y[1]), input_data))
-            y = cp.concatenate(y_parts)
+        if len(input_data) > 1:
```

Comment:
input_utils.concatenate nicely handles the len(lst) == 0 case internally so you could eliminate the if altogether and just use this branch.